chain/client-primitives/src/types.rs

```diff
@@ -153,6 +153,18 @@ impl SyncStatus {
             _ => true,
         }
     }
+
+    pub fn cast_to_int(&self) -> u8 {
+        match self {
+            SyncStatus::AwaitingPeers => 0,
+            SyncStatus::NoSync => 1,
```

Comment:
It might make more sense to make `SyncStatus::NoSync` be 0 since we expect most nodes are at NoSync